main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -1138,7 +1139,52 @@ object Typer {
         } yield (matchConstrs.flatten ++ ruleBodyConstrs.flatten, resultTyp, resultPur, resultEff)
 
 
-      case KindedAst.Expression.RestrictableChoose(star, exps0, rules0, tvar, loc) => ??? // TODO RESTR-VARS
+      case KindedAst.Expression.RestrictableChoose(true, exp0, rules0, tvar, loc) => ??? // TODO RESTR-VARS
+
+      case KindedAst.Expression.RestrictableChoose(false, exp0, rules0, tvar, loc) =>
```

Comment:
Proposal: Lets start breaking the typer a little bit a part. You could introduce `unification/RestrictableChoose.scala`. I know its not technically unification, but might be nice to group the type rules like that.

Thoughts?

Reply:
Yes. I have put it into `RestrictableChooseInference`.

I'm actually wondering if type inference as a whole should be moved into its own file/directory.

Reply:
Lets create an inference package and put `RestrictableChoose` into it.